### PR TITLE
[HVAC] Handle null BuiltIn field on Preset write according to spec

### PR DESCRIPTION
**Pull request by hasty**

### CODE REVIEW

File: src/app/clusters/thermostat-server/thermostat-server-presets.cpp [hunk 1]
```diff
@@ -367,37 +368,52 @@ CHIP_ERROR ThermostatAttrAccess::AppendPendingPreset(Thermostat::Delegate * dele
     }
```
Comment (bzbarsky-apple) [inline src/app/clusters/thermostat-server/thermostat-server-presets.cpp:368]: Shouldn't we set BuiltIn to false on `preset` here if it's coming in with BuiltIn set to null and has no handle?  At least assuming this PR is implementing https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/10277

File: src/python_testing/TC_TSTAT_4_2.py [hunk 2]
```diff
@@ -269,7 +269,9 @@ async def test_TC_TSTAT_4_2(self):
             await self.send_atomic_request_begin_command()
 
             # Write to the presets attribute after calling AtomicRequest command
-            await self.write_presets(endpoint=endpoint, presets=new_presets)
+            test_presets = copy.deepcopy(new_presets)
```
Comment (bzbarsky-apple) [inline src/python_testing/TC_TSTAT_4_2.py:272]: We should test both cases: null on a preset matching an existing handle, and null on a preset with a null handle.

File: src/app/clusters/thermostat-server/thermostat-server-presets.cpp (hunk 1 above)
Comment (hasty) [reply src/app/clusters/thermostat-server/thermostat-server-presets.cpp:370]: Done.

File: src/python_testing/TC_TSTAT_4_2.py (hunk 2 above)
Comment (hasty) [reply src/python_testing/TC_TSTAT_4_2.py:276]: OK, added to the above test for the other case

**Review by bzbarsky-apple — APPROVED** [review]

**Review by bauerschwan — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**